src/statscollector/StatsCollector.ts

```diff
@@ -698,4 +727,34 @@ export default class StatsCollector implements RedundantAudioRecoveryMetricsObse
     }
     return null;
   }
+
+  private isVideoSourceMetricReport(rawMetricReport: RawMetricReport): boolean {
+    return rawMetricReport.type === 'media-source' && rawMetricReport.kind === 'video';
+  }
+
+  private addRenderResolutionToVideoDownstreamMetrics(
+    metricReport: StreamMetricReport,
+    ssrc: number
+  ): void {
+    const streamId = this.videoStreamIndex.streamIdForSSRC(Number(ssrc));
```

Comment:
Nit: If this isn't an expected occurrence, can we log if this, the next line, and line after that do not find a mapping?